src/rpc/service_eth.go

```diff
@@ -273,3 +275,36 @@ var DefaultProcessTransaction = func(tx *txtypes.Transaction, ctx context.Contex
 
 	return submitTx, nil
 }
+
+func (p *Processor) MonitorBalance(ctx context.Context, delayInSeconds int) {
+	timer := time.NewTicker(time.Duration(delayInSeconds) * time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			utils.Logger.Info().Msg("Stopping because context is done.")
+			return
+
+		case <-timer.C:
+			balance, err := p.Client.BalanceAt(context.Background(), *p.SigningAddress, nil)
```

Comment:
think we should use the same context passed as a param here, as you did in line 284.